src/plugins/discover/public/application/angular/doc_table/components/pager/tool_bar_pagination.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPagination,
+  EuiPopover,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n/';
+
+interface ToolBarPaginationProps {
+  pageSize: number;
+  pageCount: number;
+  activePage: number;
+  onPageClick: (page: number) => void;
+  onPageSizeChange: (size: number) => void;
+}
+
+export const ToolBarPagination = ({
+  pageSize,
+  pageCount,
+  activePage,
+  onPageSizeChange,
+  onPageClick,
+}: ToolBarPaginationProps) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const rowsWord = i18n.translate('discover.docTable.rows', {
+    defaultMessage: 'rows',
+  });
+
+  const onChooseRowsClick = () => setIsPopoverOpen((prevIsPopoverOpen) => !prevIsPopoverOpen);
+
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const getIconType = (size: number) => {
+    return size === pageSize ? 'check' : 'empty';
+  };
+
+  const items = [
+    <EuiContextMenuItem
+      key="25 rows"
+      icon={getIconType(25)}
+      onClick={() => {
+        closePopover();
+        onPageSizeChange(25);
+      }}
+    >
+      25 {rowsWord}
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="50 rows"
+      icon={getIconType(50)}
+      onClick={() => {
+        closePopover();
+        onPageSizeChange(50);
+      }}
+    >
+      50 {rowsWord}
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="100 rows"
+      icon={getIconType(100)}
+      onClick={() => {
+        closePopover();
+        onPageSizeChange(100);
+      }}
+    >
+      100 {rowsWord}
+    </EuiContextMenuItem>,
+  ];
```

Comment:
suggestion: this looks like a good candidate for a mapping function like 

`const items = [25, 50, 100].map(size => (<EuiContextMenuItem...))`

Reply:
Yes, thank you!